### PR TITLE
[Graphbolt]Fix negative sampler

### PR DESCRIPTION
**Pull request by peizhou001**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: graphbolt/src/fused_csc_sampling_graph.cc [hunk 1]
```diff
@@ -712,18 +712,6 @@ FusedCSCSamplingGraph::TemporalSampleNeighbors(
   }
 }
 
-std::tuple<torch::Tensor, torch::Tensor>
```
Comment (Rhett-Ying) [inline graphbolt/src/fused_csc_sampling_graph.cc:715]: remove declaration from header file?

File: python/dgl/graphbolt/impl/fused_csc_sampling_graph.py [hunk 2]
```diff
@@ -903,24 +903,13 @@ def sample_negative_edges_uniform(
...
-                self.node_type_offset is not None
-            ), "The 'node_type_offset' array is necessary for performing \
-                negative sampling by edge type."
-            _, _, dst_node_type = etype_str_to_tuple(edge_type)
-            dst_node_type_id = self.node_type_to_id[dst_node_type]
-            offset = self._node_type_offset_list
-            max_node_id = (
-                offset[dst_node_type_id + 1] - offset[dst_node_type_id]
-            )
+        if edge_type:
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/fused_csc_sampling_graph.py:906]: as no c++ dependency on `FusedCSCSamplingGraph` now, why not implement the negative sampling in `UniformNegativeSampler::_sample_with_etype()` directly?

File: python/dgl/graphbolt/impl/uniform_negative_sampler.py [hunk 3]
```diff
@@ -32,20 +32,23 @@ class UniformNegativeSampler(NegativeSampler):
...
-    ...     item_set, batch_size=1,)
+    ...     item_set, batch_size=4,)
     >>> neg_sampler = gb.UniformNegativeSampler(
     ...     item_sampler, graph, 2)
     >>> for minibatch in neg_sampler:
     ...       print(minibatch.negative_srcs)
     ...       print(minibatch.negative_dsts)
-    (tensor([0, 0, 0]), tensor([1, 1, 2]), tensor([1, 0, 0]))
-    (tensor([1, 1, 1]), tensor([2, 1, 2]), tensor([1, 0, 0]))
+    None
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/uniform_negative_sampler.py:47]: add description about `neg_srcs` is always `None` in our negative sampler?

File: python/dgl/graphbolt/impl/fused_csc_sampling_graph.py (hunk 2 above)
Comment (peizhou001) [reply python/dgl/graphbolt/impl/fused_csc_sampling_graph.py:906]: It need information from graph.

Comment (Rhett-Ying) [reply python/dgl/graphbolt/impl/fused_csc_sampling_graph.py:907]: graph is already passed into negative sampler

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**